### PR TITLE
feat: shortened the way c# properties are generated

### LINKED ISSUE #670: Utilising shorthands for accessors in CSHARP_DEFAULT_CLASS_PRESET
#### Reason/Context
Currently, each property is generated like this
```
private Type field;

public Type Field
{
    get { return field; }
    set { field = value; }
}
```
which generates 6 times more code than it's necessary. C# allows for such a shorthand:
```
public Type Field { get; set; }
```
Way less code, the result the same

#### Description
I don't think it's a breaking change. Generated data models will be able to be used the same way they currently are.
The resolution for it would include modifying the `src/generators/csharp/renderers/ClassRenderer.ts` file and more specifically the implementation of CSHARP_DEFAULT_CLASS_PRESET. If anybody would ever want to use the more complex way of declaring properties, they would still be able to do it by implementing their own CsharpClassPreset, but I don't see a reason to make it default.

### PR DESCRIPTION
**Pull request by slowikowskiarkadiusz**
**Description**

- Incorporates using shorthand for property & accessors generation for C#
- Some formatting in awkward places

Code generation output before my changes
```
public class ExampleObject {
  private float? numberField;
  private dynamic[] collectionField;
  private Dictionary<string, dynamic> additionalProperties;

  public float? NumberField 
  {
    get { return numberField; }
    set { numberField = value; }
  }

  public dynamic[] CollectionField 
  {
    get { return collectionField; }
    set { collectionField = value; }
  }

  public Dictionary<string, dynamic> AdditionalProperties 
  {
    get { return additionalProperties; }
    set { additionalProperties = value; }
  }
}
```

Code generation output after my changes
```
public class ExampleObject {
  public float? NumberField { get; set; }
  public dynamic[] CollectionField { get; set; }
  public Dictionary<string, dynamic> AdditionalProperties { get; set; }
}
```

**Related issue(s)**

Resolved #670

### CODE REVIEW

**Comment by jonaslagoni** [comment]
While I do think this is great to have, I don't think we should overwrite the default behavior, because it's only a matter of preference.

Instead, I suggest you add it to options such as `useShortHandAccessors` or something similar. And then based on that you either include either one of them as needed.

Also make sure to add examples, tests and documentation 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
While I do think this is great to have, I don't think we should overwrite the default behavior, because it's only a matter of preference.

Instead, I suggest you add it to options such as useShortHandAccessors or something similar. And then based on that you either include either one of them as needed.

**Comment by slowikowskiarkadiusz** [comment]
Alright, so I've made a possibility to alter the default behaviour of CSHARP_COMMON_PRESET by setting an 'autoImplementedProperties' option to true, but I'm not sure whether I like it. It required me to pass options through a couple of functions and it basically completely changes the way the preset behaves. I wonder whether it would make more sense to just implement it as a separate preset instead. What do you think, @jonaslagoni?

File: test/generators/csharp/presets/__snapshots__/CommonPreset.spec.ts.snap
Comment (jonaslagoni) [inline test/generators/csharp/presets/__snapshots__/CommonPreset.spec.ts.snap:178]: Should this not be short accessors? 🤔

File: test/generators/csharp/presets/CommonPreset.spec.ts
Comment (jonaslagoni) [inline test/generators/csharp/presets/CommonPreset.spec.ts:64]: Don't think we really need this one, as the example is plenty IMO 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
You have a few linting errors 🙂 

> Alright, so I've made a possibility to alter the default behaviour of CSHARP_COMMON_PRESET by setting the 'autoImplementedProperties' option to true, but I'm not sure whether I like it. It required me to pass options through a couple of functions and it basically completely changes the way the preset behaves. I wonder whether it would make more sense to just implement the whole thing as a separate preset instead. What do you think, @jonaslagoni?

Well, as presets are still in the early days, it's hard to say if they should be in one or the other. So up to you. I am happy with where they are at the moment 🙂

File: test/generators/csharp/presets/__snapshots__/CommonPreset.spec.ts.snap
Comment (slowikowskiarkadiusz) [reply test/generators/csharp/presets/__snapshots__/CommonPreset.spec.ts.snap]: Should be, but now it's removed anyways :)

File: test/generators/csharp/presets/CommonPreset.spec.ts
Comment (slowikowskiarkadiusz) [reply test/generators/csharp/presets/CommonPreset.spec.ts]: Done :)

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
@slowikowskiarkadiusz looks good to me 👍 You need to resolve some conflicts from master though, i.e. try merging the upstream master into your branch 🙂

**Comment by jonaslagoni** [comment]
@ron-debajyoti anything you want to add?

**Comment by ron-debajyoti** [comment]
I resolved the merge conflict, but looks like a test is failing. Mind taking a look ?

**Comment by jonaslagoni** [comment]
@slowikowskiarkadiusz do you mind updating the test snapshots so we can merge this? 🙂

**Comment by slowikowskiarkadiusz** [comment]
> @slowikowskiarkadiusz do you mind updating the test snapshots so we can merge this? 🙂

Sorry for the delay :) Done. All of the tests pass locally for me

**Comment by slowikowskiarkadiusz** [comment]
> Looks like linter is having some problems as well 🙂

Should be in place now

**Review by jonaslagoni — APPROVED** [review]

**Comment by jonaslagoni** [comment]
@ron-debajyoti as codeowner you have final review 🙂

**Review by ron-debajyoti — APPROVED** [review]

**Comment by jonaslagoni** [comment]
@all-contributors please add @slowikowskiarkadiusz for code, test, example, doc

**Pull request merged**
